File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/FcmEventsListener.java
package com.jdroid.android.firebase.fcm;

import com.google.firebase.messaging.RemoteMessage;

public interface FcmEventsListener {

	void onMessageReceived(RemoteMessage remoteMessage);

	void onMessageSent(String msgId);

	void onSendError(String msgId, Exception exception);

	void onDeletedMessages();

	void onNewToken(String token);
}

File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/FcmEventsListener.kt
package com.jdroid.android.firebase.fcm

import com.google.firebase.messaging.RemoteMessage

interface FcmEventsListener {

    fun onMessageReceived(remoteMessage: RemoteMessage)

    fun onMessageSent(msgId: String)

    fun onSendError(msgId: String, exception: Exception)

    fun onDeletedMessages()

    fun onNewToken(token: String)
}
